crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -164,9 +164,11 @@ impl BaseClient {
         let mut ambiguity_cache = AmbiguityCache::new(store.inner.clone());
 
         let account_data = &extensions.account_data;
-        if !account_data.is_empty() {
-            self.handle_account_data(&account_data.global, &mut changes).await;
-        }
+        let global_account_data_events = if !account_data.is_empty() {
+            self.handle_account_data(&account_data.global, &mut changes).await
+        } else {
+            Vec::new()
```

Comment:
Note: Here this is fine to get a `Vec::new()`, it does zero allocation.

I know it's different from what we've talked in the above comments, but I'm sharing some Rust knowledge.